### ISSUE TITLE
refactor(Label): update props to the latest specs

### ISSUE THREAD

**Issue by levithomason** [Semantic-Org/Semantic-UI-React#474]
**Shorthand**

Currently, the Label allows shorthand props `icon`, `image`, and `text` along with `children`.  These should be `disallow()`ed via `propTypes` and early returns put in place to render one or the other.

**`text`**

This prop should be renamed to `content`.  See #391.

**`link`**

This prop renders as an `a` tag.  It should go away as we now have augmentation `<Label as='a' />` instead.

**`empty`**

Currently, the Label auto applies the `empty` className when there are no children.  Magic is bad, unless it is magic application of something that is _always required_, like a [Checkbox fitted prop](http://technologyadvice.github.io/stardust/modules/checkbox#fitted).  A Label with no children does not _always require_ the `empty` class.  It is a slightly larger but valid Label without it:

![image](https://cloud.githubusercontent.com/assets/5067638/18392520/f0bfb200-7666-11e6-9124-be94038f1c6f.png)

We should remove the auto application of this className.

**Docs & Tests**

Docs and tests should then be updated for all usages.

**Comment by jeffcarbs** [comment]
Somewhat related - I'm having an issue with this line which adds the `empty` class: https://github.com/TechnologyAdvice/stardust/blob/master/src/elements/Label/Label.js#L41.

I'm trying to set a circular label with the `text` shorthand prop and it's getting the `empty` class. This is because that logic is only based on children: 

``` js

// Currently
<Label text={count} circular />
<div class="ui empty circular label">11</div>

// To make it non-empty, needs children
<Label circular>{count}</Label>
<div class="ui circular label">11</div>
```

I could see `empty` being set when there's no content of any kind, although I think I'd prefer to just set it manually:

``` js
// Better (but still not completely intuitive):
<Label circular />
<div class="ui empty circular label"></div>

// Ideal:
<Label empty circular />
<div class="ui empty circular label"></div>
```

**Comment by levithomason** [comment]
Yep, that's bad.  See https://github.com/TechnologyAdvice/stardust/issues/485

**Comment by levithomason** [comment]
Actually, let's just do this all at once, updated description above.

**Comment by levithomason** [comment]
Fixed in #486
